Purchases/Public/PurchaserInfo.swift

```diff
@@ -294,9 +294,8 @@ private extension PurchaserInfo {
 
     func parseDatesIn(transactionsByProductId: [String: [String: Any]], dateLabel: String) -> [String: Date?] {
         // mapValues will keep the key-value pair in the dictionary for nil values, as desired
-        return transactionsByProductId.mapValues { maybeTransaction in
-            if let transactionFieldsByKey = maybeTransaction as? [String: String],
-               let dateString = transactionFieldsByKey[dateLabel] {
+        return transactionsByProductId.mapValues { transaction in
+            if let dateString = transaction[dateLabel] as? String {
```

Comment:
context for other readers: 
`maybeTransaction` here wasn't necessary, since the transaction would always be there - we're just mapping values from `transactionsByProductId`. 
As mentioned in the PR description, the transaction dict does have values that aren't string, like `is_sandbox` (bool), and nullable elements, so the cast would always fail. `transaction` is already `[String: Any]`, so no further casting is needed.